src/brs/BlockchainProcessorImpl.java

```diff
@@ -703,8 +703,12 @@ private void processFork(Peer peer, final List<Block> forkBlocks, long forkBlock
             break;
           }
         } catch (Exception exception) {
-          exception.printStackTrace();
-          logger.error("Uncaught exception in blockImporterThread", exception);
+          if (exception.toString().contains("[SQLITE_BUSY]") || exception.toString().contains("[SQLITE_BUSY_SNAPSHOT]")) {
```

Comment:
Hey, I'm not really sure about this, because, when you look at the code, it is apparent that we need an exception to occur there.

https://github.com/signum-network/signum-node/blob/85561beaae0ebd7a5c178520049f235f68720cba/src/brs/BlockchainProcessorImpl.java#L1145-L1155

Exception occurs in this file at line 1150 in `addBlock()`, in a call actually down in the hierarchy, and is caught up in the call stack hierarchy, at the place where I put my patch.

The thing is: the exception allows addBlock() to terminate early, so the block is not removed at line 1155 by (`downloadCache.removeBlock(block);`). This way, it is naturally processed again later by the thread, and ends up in the database at a later time.

So if I still want this function to terminate early thanks to an exception, and if I handle the exception thanks to an ExecuteListener like you suggest, I can probably silence the SQLITE_BUSY, but I'd still need to raise something. Otherwise the code above is going to continue further down the `addBlock()`, which we do not want.
So we'll still end up with a stacktrace where I put my original patch.

Reply:
I'll try to think again about this later, but just leave it aside for now if you need to publish 3.8.3.